src/ipyaladin/region_converter.py

```diff
@@ -0,0 +1,283 @@
+    PolygonSkyRegion = None
+    Region = None
+    CircleSkyRegion = None
+    EllipseSkyRegion = None
+    LineSkyRegion = None
+from astropy.coordinates import SkyCoord
+from typing import Union
+
+TWICE_PI = 2 * math.pi
+
+
+class RefToLocalRotMatrix:
+    """A rotation matrix from the reference frame to the local frame.
+
+    The reference frame is the ICRS frame. The local frame is
+    defined by the center of the local reference frame.
+
+    Attributes
+    ----------
+    r11-r33 : float
+        The elements of the rotation matrix.
+
+    """
+
+    def __init__(
+        self,
+        rotation_matrix: np.ndarray,
+    ) -> None:
+        self.rotation_matrix = rotation_matrix
+
+    @classmethod
+    def from_center(cls: any, lon: float, lat: float) -> "RefToLocalRotMatrix":
+        """Create a rotation matrix from the center of the local reference frame.
+
+        Parameters
+        ----------
+        lon : float
+            The longitude of the center.
+        lat : float
+            The latitude of the center.
+
+        Returns
+        -------
+        RefToLocalRotMatrix
+            The rotation matrix.
+
+        """
+        ca, sa = math.cos(lon), math.sin(lon)
+        cd, sd = math.cos(lat), math.sin(lat)
+        return cls(
+            np.array([[ca * cd, sa * cd, sd], [-sa, ca, 0.0], [-ca * sd, -sa * sd, cd]])
+        )
+
+    def to_global_xyz(self, x: float, y: float, z: float) -> tuple[float, float, float]:
+        """Convert local cartesian coordinates to global cartesian coordinates.
+
+        Parameters
+        ----------
+        x : float
+            The x coordinate.
+        y : float
+            The y coordinate.
+        z : float
+            The z coordinate.
+
+        Returns
+        -------
+        tuple[float, float, float]
+            The x, y, and z coordinates.
+
+        """
+        rotated_matrix = np.sum(
+            self.rotation_matrix * np.array([[x], [y], [z]]), axis=0
+        )
+        return rotated_matrix[0], rotated_matrix[1], rotated_matrix[2]
+
+    def to_global_coo(self, x: float, y: float, z: float) -> tuple[float, float]:
+        """Convert local cartesian coordinates to global spherical coordinates.
+
+        Parameters
+        ----------
+        x : float
+            The x coordinate.
+        y : float
+            The y coordinate.
+        z : float
+            The z coordinate.
+
+        Returns
+        -------
+        tuple[float, float]
+            The longitude and latitude.
+
+        """
+        x, y, z = self.to_global_xyz(x, y, z)
+        # Convert cartesian coordinates to spherical coordinates.
+        r2 = x * x + y * y
+        lat = math.atan2(z, math.sqrt(r2))
+        lon = math.atan2(y, x)
+        if lon < 0:
+            lon += TWICE_PI
+        return lon, lat
+
+
+def rectangle_to_polygon_region(region: RectangleSkyRegion) -> PolygonSkyRegion:
+    """Convert a RectangleSkyRegion to a PolygonSkyRegion.
```

Comment:
maybe specify that this code is a transcription of cdshealpix: https://github.com/cds-astro/cds-healpix-rust/blob/master/src/nested/mod.rs#L3687-3742